lib/Util/Notices/UnauthorizedEmailNotice.php

```diff
@@ -73,9 +77,13 @@ private function getSettingsButtons($validation_error) {
   }
 
   private function getAuthorizationLink($validation_error) {
-    if (!empty($validation_error['invalid_sender_address']) && !empty($validation_error['invalid_confirmation_address'])) {
-      $authorize_link = $this->wp->x('Authorize %email and %email2', 'Link for user to authorize their email address');
-      $authorize_link = str_replace('%email', EscapeHelper::escapeHtmlText($validation_error['invalid_sender_address']), $authorize_link);
+    if (
+      !empty($validation_error['invalid_sender_address'])
+      && !empty($validation_error['invalid_confirmation_address'])
+      && $validation_error['invalid_sender_address'] !== $validation_error['invalid_confirmation_address']
+    ) {
+      $authorize_link = $this->wp->x('Authorize %email1 and %email2', 'Link for user to authorize their email address');
+      $authorize_link = str_replace('%email1', EscapeHelper::escapeHtmlText($validation_error['invalid_sender_address']), $authorize_link);
```

Comment:
Looks good. However (and I notice that just now), considering `UnauthorizedEmailNotice.php` is a completely new class I don't think we should use "hacky" `str_replace('%email1', ...)`. PHP has `sprintf` exactly for this. Would you mind adding a commit to fix that?

Reply:
I used str_replace to keep nicer placeholders for translators.